src/sql/src/plan/statement/ddl.rs

```diff
@@ -5012,13 +5093,61 @@ pub fn describe_alter_object_swap(
 }
 
 pub fn plan_alter_object_swap(
-    _: &mut StatementContext,
-    _: AlterObjectSwapStatement,
+    scx: &mut StatementContext,
+    stmt: AlterObjectSwapStatement,
 ) -> Result<Plan, PlanError> {
-    Err(PlanError::Unsupported {
-        feature: "ALTER ... SWAP ...".to_string(),
-        issue_no: Some(12972),
-    })
+    scx.require_feature_flag(&vars::ENABLE_ALTER_SWAP)?;
+
+    let AlterObjectSwapStatement {
+        object_type,
+        name_a,
+        name_b,
+    } = stmt;
+    let object_type = object_type.into();
+
+    // We'll try 10 times to generate a temporary suffix.
+    let gen_temp_suffix = |check_fn: &dyn Fn(&str) -> bool| {
+        let mut attempts = 0;
+        let name_temp = loop {
+            attempts += 1;
+            if attempts > 10 {
+                tracing::warn!("Unable to generate temp id for swapping");
+                sql_bail!("unable to swap!");
+            }
+
+            // To make these temporary names a bit more manageable, we make them short, by using
+            // the last component of a UUID, which should be 12 characters long.
+            //
+            // Note: the reason we use the last 12 characters is because the bits 6, 7, and 12 - 15
+            // are all hard coded <https://www.rfc-editor.org/rfc/rfc4122#section-4.4>.
+            let temp_uuid = uuid::Uuid::new_v4().as_hyphenated().to_string();
+            let short_id: String = temp_uuid.chars().rev().take_while(|c| *c != '-').collect();
+
+            // Call the provided closure to make sure this name is unique!
+            if check_fn(&short_id) {
```

Comment:
Is there a chance of another schema/cluster with this name being generated meanwhile or is it locked already?